opt_einsum/tests/test_parser.py

```diff
@@ -12,3 +13,21 @@ def test_get_symbol():
     assert get_symbol(55295) == "\ud88b"
     assert get_symbol(55296) == "\ue000"
     assert get_symbol(57343) == "\ue7ff"
+
+
+def test_parse_einsum_input():
+    eq = "ab,bc,cd"
+    ops = [np.random.rand(2, 3), np.random.rand(3, 4), np.random.rand(4, 5)]
+    input_subscripts, output_subscript, operands = parse_einsum_input([eq, *ops], shapes=True)
```

Comment:
My understanding is that this test shouldn't work as we expect tuples of ints rather than tensors. I must be missing something here however.

Reply:
Oh that is a very good point.

This also just works because parse_einsum_input doesn't do anything to the operands (whether they be shapes or operands) and should just pass them along.

Reply:
I've modified the test so it tests the intended behavior.

Reply:
We have a faulty test in this case, It just so happens that `len(arr)` provides the shape of the first dimension. We need to ensure that a corresponding test fails for the given test case. A fairly minor point, but the tests as written can be misleading in the future.